### DISCUSSION TITLE
Chat templates and llama-server

### DISCUSSION THREAD

**Discussion by tesseract241** [ggml-org/llama.cpp#9741] [Q&A]
Hello all, I'm trying to wrap my mind around how to use llama-server.
The chat template wiki page says  
>Currently, it's not possible to use your own chat template with llama.cpp server's /chat/completions
>One of the possible solutions is use /completions endpoint instead, and write your own code (for example, using python) to apply a template before passing the final prompt to /completions

I'm not sure how to read this: are chat templates embedded into a gguf file used if present?
Let's assume the model has no chat template and I want to provide my own, should I still separate the system prompt (wrapped in its special tokens) from the user prompt (also wrapped in its special tokens and ending with the token that starts the AI reply) or just put everything in the user prompt instead?

**Comment by jurov** [comment]
It's not separate, there's just one prompt for everything. Put the system one first, then append user prompt, then append the conversation. 

There's some prompt information in gguf files and it is shown in console output on startup but don't know if it can be used automatically. I just manually copied it.
